src/main/resources/mybatis/groundwaterDailyValueWaterML.xml

```diff
@@ -14,13 +14,32 @@
                        observered_property_reference,
                        statistic,
                        statistic_reference,
-                       xmlagg(xmlelement(name "wml2:point", 
-                          xmlelement(name "wml2:MeasurementTVP", 
-                            xmlelement(name "wml2:time",
-                              to_char(time_step at time zone 'UTC',
-                                      'YYYY-MM-DD')),
-                            xmlelement(name "wml2:value", result)
-                       ))) result_agg
+                       unit_of_measure,
+                       unit_of_measure_reference,
+                       xmlagg(xmlelement(name "wml2:point",
+                         xmlelement(name "wml2:MeasurementTVP",
+                           xmlelement(name "wml2:time",
+                             to_char(time_step at time zone 'UTC',
+                                     'YYYY-MM-DD')),
+                           xmlelement(name "wml2:value", result),
+                           xmlelement(name "wml2:metaData",
+                             xmlelement(name "wml2:TVPMeasurementMetadata",
+                               case
+                                 when qualifiers is null then
+                                   xmlelement(name "wml2:qualifier")
+                                 else
+                                   (select xmlagg(xmlelement(name "wml2:qualifier", qualifier))
+                                       from jsonb_array_elements_text(qualifiers) as qualifier)
+                               end,
+                               case
+                                 when approvals is null then
+                                   xmlelement(name "wml2:status")
+                                 else
+                                   (select xmlagg(xmlelement(name "wml2:status", approval))
+                                       from jsonb_array_elements_text(approvals) as approval)
+                               end
+                           ))
+                       ))) point_agg
```

Comment:
nicely done here. I read every line and followed the code. I presume it works as I read it.